doc/sphinx/proofs/writing-proofs/reasoning-inductives.rst

```diff
@@ -1001,90 +911,61 @@ Induction
 
 .. seealso:: :tacn:`functional inversion`
 
-.. todo move the following label to just
-        before ":tacn:`discriminate` and :tacn:`injection`"
-
-.. _equality-inductive_types:
-
-Equality and inductive sets
----------------------------
-
-We describe in this section some special purpose tactics dealing with
-equality and inductive sets or types. These tactics use the
-equality :g:`eq:forall (A:Type), A->A->Prop`, simply written with the infix
-symbol :g:`=`.
-
 .. tacn:: decide equality
-   :name: decide equality
 
-   This tactic solves a goal of the form :g:`forall x y : R, {x = y} + {~ x = y}`,
+   .. todo PR: why the curly braces below instead of parentheses?
+      shouldn't "+" be "/\"?
+
+   Solves a goal of the form :g:`forall x y : R, {x = y} + {~ x = y}`,
    where :g:`R` is an inductive type such that its constructors do not take
    proofs or functions as arguments, nor objects in dependent types. It
    solves goals of the form :g:`{x = y} + {~ x = y}` as well.
 
-.. tacn:: compare @term @term
-   :name: compare
+.. tacn:: compare @one_term__1 @one_term__2
 
-   This tactic compares two given objects :n:`@term` and :n:`@term` of an
-   inductive datatype. If :g:`G` is the current goal, it leaves the sub-
-   goals :n:`@term =@term -> G` and :n:`~ @term = @term -> G`. The type of
-   :n:`@term` and :n:`@term` must satisfy the same restrictions as in the
+   Compares two :n:`@one_term`\s of an
+   inductive datatype. If :g:`G` is the current goal, it leaves the
+   sub-goals :n:`@one_term__1 = @one_term__2 -> G` and :n:`~ @one_term__1 = @one_term__2 -> G`.
+   The type of the :n:`@one_term`\s must satisfy the same restrictions as in the
    tactic ``decide equality``.
 
-.. tacn:: simplify_eq @term
-   :name: simplify_eq
+.. tacn:: simplify_eq {? @induction_arg }
+
+   .. todo PR: seems like it requires term1 <> term2
+
+   Examines goals that have the form :n:`@term__1 = @term__2`.  If the terms are
+   structurally different, the tactic does a :tacn:`discriminate`.  Otherwise, it does
+   an :tacn:`injection` to simplify the equality, if possible.
 
-   Let :n:`@term` be the proof of a statement of conclusion :n:`@term = @term`.
-   If :n:`@term` and :n:`@term` are structurally different (in the sense
-   described for the tactic :tacn:`discriminate`), then the tactic
-   ``simplify_eq`` behaves as :n:`discriminate @term`, otherwise it behaves as
-   :n:`injection @term`.
+   See the description of :token:`induction_arg` for an explanation of the
+   parameters.
+
+   .. todo PR: is there a better name than "induction_arg"?
```

Comment:
What about `ident_nat_or_term` or something along these lines that would simply describe the usual cases it encompasses?

Reply:
It's not just the name.  I think the main issue is that the text describing `induction_arg` is too specific to `destruct` and AFAIK simply subsituting `simplify_eq` into that text is not nearly enough.  So seems like it needs a new description entirely and we can pick a new name for the nonterminal.

How do we describe it for here?

Reply:
> simply subsituting `simplify_eq` into that text is not nearly enough

Should be checked but there are relatively good chances that this is actually enough. Currently, the explanations for `induction_arg` as repeated in `destruct`, `induction`, `injection`... and it would indeed probably be worth factorizing them outside all of these tactics, using general language and making sure that we don't miss any subcase. Then we could test that it does indeed behave like it is described. (In fact, if this is too much for the manual, we could at least use the documentation to systematically extend the test-suite to cover all documented cases).

Reply:
I'll leave this unchanged for now.